crates/sqlexec/src/planner/physical_plan/install.rs

```diff
@@ -0,0 +1,166 @@
+        )))
+    }
+
+    fn statistics(&self) -> DataFusionResult<Statistics> {
+        Ok(Statistics::new_unknown(self.schema().as_ref()))
+    }
+}
+
+impl DisplayAs for InstallExec {
+    fn fmt_as(&self, _t: DisplayFormatType, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "InstallExec")
+    }
+}
+
+impl InstallExec {
+    async fn install_extension(self) -> DataFusionResult<RecordBatch> {
+        let output_schema = self.schema();
+        let extension_dir = get_extension_path();
+        let extension = self.extension;
+
+        if let Some(path) = normalize_extension_name(&extension) {
+            let mut ext_name = path.file_name().and_then(|s| s.to_str()).ok_or_else(|| {
+                DataFusionError::Execution(format!("Failed to get file name from path: {:?}", path))
+            })?;
+            if ext_name.starts_with(DLL_PREFIX) {
+                ext_name = &ext_name[DLL_PREFIX.len()..];
+            }
+            if ext_name.ends_with(DLL_SUFFIX) {
+                ext_name = &ext_name[..ext_name.len() - DLL_SUFFIX.len()];
+            }
+
+            let dest = extension_dir.join(ext_name);
+
+            if !dest.exists() {
+                fs::create_dir_all(&extension_dir).map_err(|e| {
+                    DataFusionError::Execution(format!(
+                        "Failed to create extension directory: {:?}",
+                        e
+                    ))
+                })?;
+                fs::copy(&path, dest).map_err(|e| {
+                    DataFusionError::Execution(format!(
+                        "Failed to copy extension to directory: {:?}",
+                        e
+                    ))
+                })?;
+            }
+            let ext_arr = GenericStringArray::<i32>::from(vec![Some(ext_name.to_string())]);
+            let batch = RecordBatch::try_new(output_schema, vec![Arc::new(ext_arr)])?;
+            Ok(batch)
+        } else {
+            Ok(RecordBatch::new_empty(output_schema))
+        }
+    }
+}
+
+/// returns some path if the extension name is valid and the file exists
+pub(super) fn normalize_extension_name(name: &str) -> Option<PathBuf> {
+    let mut maybe_path = Path::new(&name).to_path_buf();
+    if !maybe_path.ends_with(DLL_SUFFIX) {
+        maybe_path.set_extension(DLL_EXTENSION);
+    };
+    std::fs::canonicalize(maybe_path).ok()
+}
+
+
+pub(super) fn get_home_dir() -> PathBuf {
+    match env::var("HOME") {
+        Ok(path) => PathBuf::from(path),
+        Err(_) => match env::var("USERPROFILE") {
+            Ok(path) => PathBuf::from(path),
+            Err(_) => panic!("Failed to get home directory"),
+        },
+    }
+}
+
+// ~/.glaredb/extensions/0.1.0/macos_x86_64
+pub(super) fn get_extension_path() -> PathBuf {
+    let mut home_dir = get_home_dir();
+    home_dir.push(".glaredb");
+    home_dir.push("extensions");
+    home_dir.push(env!("CARGO_PKG_VERSION"));
+    home_dir.push(format!("{}_{}", FAMILY, ARCH));
+    home_dir
+}
```

Comment:
i purposely didn't use the `data-dir` because installed extensions should be global. We are doing something similar with the `.history` file. Those are system wide.

Reply:
I'm not convinced that there should be _any_ global state (history, extensions, etc.) but regardless of what directory we choose or how we choose to implement that, we shouldn't be reading from the home directory or joining the path together at _in the execution plan_ and do it once/only once per process and add it to the appropriate context.

Reply:
I think this is an exception as we want exactly one copy of that extension per glaredb version. The functionality set forth closely follows the functionality of duckdb's extension system. They are installed globally, and loaded on a per session basis.

We can move this variable to a static/const or even a user configurable session_variable.

Reply:
moved to a configurable session_var `extension_directory`